frontend/internal-packages/agent/src/chat/workflow/services/workflow.ts

```diff
@@ -60,23 +68,34 @@ const createGraph = () => {
     .addEdge('prepareDML', 'validateSchema')
     .addEdge('finalizeArtifacts', END)
 
+    // Conditional edges with retry logic - each node will retry up to maxRetries times on error
+    // If maxRetries is exceeded and error persists, workflow goes to END
```

Comment:
https://langchain-ai.github.io/langgraphjs/how-tos/node-retry-policies/

I found it a bit hard to read, so I checked if LangChain has a built-in retry system, and it seems that you can set a retryPolicy on addNode. It may be better to match this one.

Migration could possibly solve the following problems: https://github.com/liam-hq/liam/pull/2167/files#r2163037269

There is no problem to merge once to solve this issue!